pkg/volume/csi/csi_plugin.go

```diff
@@ -914,3 +920,23 @@ func highestSupportedVersion(versions []string) (*utilversion.Version, error) {
 	}
 	return highestSupportedVersion, nil
 }
+
+// waitForAPIServerForever waits forever to get the APIServer Version as a proxy
+// for a healthy APIServer.
+func waitForAPIServerForever(client clientset.Interface) error {
+	var lastErr error
+	err := wait.PollInfinite(time.Second, func() (bool, error) {
+		_, lastErr = client.Discovery().ServerVersion()
+		if lastErr != nil {
+			lastErr = fmt.Errorf("failed to get apiserver version: %v", lastErr)
+			return false, nil
+		}
+
+		return true, nil
+	})
+	if err != nil {
```

Comment:
It seems the err would always be nil (due to nil error being returned from the lambda)

Reply:
true. looks like line 932 should return lastErr

Reply:
No, `lastErr` must not be returned from inside `PollInfinite` argument - it would stop polling after the first `ServerVersion()` error. The whole point of `PollInfinite` is to loop while `ServerVersion()` returns error and stop when it returns nil.

Reading `PollInfinite`, it should never return error in the current implementation, but better to be on the safe side.

Reply:
I'm assuming the intent here is to not return an error if the initialization completes within the configured backoff internal. Why are errors not logged though at a low severity? Otherwise transitory errors are hidden when troubleshooting.